HomeWork2/DmytroFenevHW2.cpp

```diff
@@ -0,0 +1,159 @@
+  case 1:
+    n = 1; //число цифр
+    std::cout << "\tEnter an integer" << std::endl;
+    std::cin >> x;
+    xp = x; //сохранение х после подсчета количества цифр
+    while ((x /= 10) > 0)
+      n++;
+    std::cout << "\tNumber of digits=" << n << std::endl;
+    y = xp / 10;
+    y1 = xp % 10; // number of low order
+    sum = y1;
+    for (int ii = 1; ii < n; ii++) {
+      yp = y / 10;
+      y1 = y % 10;
+      sum = sum + y1; //сумма цифр
+      y = yp;
+    }
+    std::cout << "\tsum=" << sum << std::endl;
+    std::cout << "\tAverage number =" << sum / n << std::endl;
+    break;
+  case 2:
+    n = 6; //число цифр
+    np = 1;
+    std::cout << "\tEnter six digits" << std::endl;
+    std::cin >> x;
+    y = x; //сохранение х после подсчета количества цифр
+    while ((x /= 10) > 0)
+      np++;
+    std::cout << np << std::endl;
+    while ((np < n) || (np > n)) {
+      np = 1;
+      std::cout << "\tEnter not six  digits" << std::endl;
+      std::cin >> x;
+      y = x;
+      while ((x /= 10) > 0)
+        np++;
+    }
+    sum1 = 0;
+    sum2 = 0;
+    for (int ii = 1; ii <= n; ii++) {
+      if (ii < 4) {
+        yp = y / 10;
+        y1 = y % 10;
+        sum1 = sum1 + y1;
+        y = yp;
+      } else {
+        yp = y / 10;
+        y1 = y % 10;
+        sum2 = sum2 + y1;
+        y = yp;
+      }
+    }
+    if (sum1 == sum2)
+      std::cout << "\tAmount match" << std::endl;
+    else
+      std::cout << "\tAmounts do not match" << std::endl;
+    break;
+  case 3:
+    // signed long l,xp;
+    int L;
+    std::cin >> L;
+    n = 1;
+    xp = abs(L);
+    xp1 = xp; //сохранение х после подсчета количества цифр
+    while ((xp1 /= 10) > 0)
+      n++;
+    y = xp;
+    for (int ii = 0; ii < n; ii++) {
+      yp = y / 10;
+      y1 = y % 10;
+      sample[ii] = y1;
+      y = yp;
+    }
+    for (int i = 0; i < n; i++) {
+      std::cout << "sample=" << sample[i] << std::endl;
+    }
+    k3 = (pow(10, n));
+    k2 = int(k3);
+    std::cout << k2 << std::endl;
+    l2 = 0;
+    for (int i = 0; i < n; i++) {
+      k2 = k2 / 10;
+      std::cout << k2 << std::endl;
+      l2 = l2 + k2 * sample[i];
+    }
+    if (L < 0)
+      l2 = -l2;
+    // l1 = sample[0] * 100 + sample[1] * 10 + sample[2];
+    std::cout << "number=" << l2 << std::endl;
+    break;
+  case 4:
+    sum = 0;
+    nc = 0;
+    std::cout << "\tEnter an integer in range...1...50" << std::endl;
+    std::cin >> nElement;
+    while ((nElement < Min) || (nElement > Max)) {
+      std::cout << "\tEnter an integer again in range 1...50" << std::endl;
+      std::cin >> nElement;
+    }
+    for (int i = 0; i < nElement; i++) {
+      std::cout << "\tEnter an integer in range...-60...+90" << std::endl;
+      std::cin >> chislo;
+      while ((chislo < cmin) || (chislo > cmax)) {
+        std::cout << "\tEnter an integer again in range -60...+90" << std::endl;
+        std::cin >> chislo;
+      }
+      (chislo % 2 == 0) ? nc++ : sum += chislo;
```

Comment:
это некоректное применение тернарного оператора. и зачем вам нужна переменая nc?
```suggestion
     sum += (chislo % 2 == 0) ? 0 :  chislo;
```

Reply:
if(chislo % 2 != 0) 
                sum += chislo;